studiopress/genesis-custom-blocks
# Add a JS filter for a preview other than the <ServerSideRender>

#### Changes
* This allows GCB to be used in [headless](https://github.com/kienstra/headless-blocks)
* Now, users can define their own React preview component
* But even if this filter isn't used much, it probably won't hurt to add it

#### Testing instructions
1. For this PR, do `composer install && npm install && npm run dev`
2. Upload and activate this plugin on `/wp-admin/plugin-install.php`: 
[headless-blocks.0.1.0.zip](https://github.com/studiopress/genesis-custom-blocks/files/6058916/headless-blocks.0.1.0.zip)
3. That [plugin](https://github.com/kienstra/headless-blocks) is the 'headless connector' for GCB, and will add alternate previews
4. `/wp-admin` > Posts > Add New
5. Add an 'Email Opt In' block that `headless-blocks` registered
6. Populate the fields: 
![the-block-in-g](https://user-images.githubusercontent.com/4063887/109447440-4b352300-7a09-11eb-98ee-7bbd151984ec.png)
7. Click outside the block so the preview shows
8. Expected: The preview is a React component, using the new JS filter `genesisCustomBlocks.alternatePreview`:  
![the-preview](https://user-images.githubusercontent.com/4063887/109447549-8d5e6480-7a09-11eb-9ddb-90743fb98505.png)
